app/domain/errors.rb

```diff
@@ -70,6 +70,11 @@ module Authentication
       code: "CONJ00017E"
     )
 
+    MissingResourceRestrictionsValue = ::Util::TrackableErrorClass.new(
```

Comment:
Line 19: In "    InsufficientPasswordComplexity" .... ", and 1 special character"

line 32 : In " AuthenticatorNotFound", is this Conjur only? What about DAP? Also, I would say "... not configured/installed/enabled" but not "not implemented..."

Lines 58-59: "Resource type '{0}' is not a supported resource restriction. " \
            "The supported resources are '{1}'",	            "The supported resources are '{1}'",
Do you mean "The supported resource restrictions are....", or "The supported restrictions are..."

Line 47: "Status check not implemented for authenticator"  What does "not implemented" mean? How about "Status of [authn-name] authenticator not verified."

Line 53 - what does this mean, in plain english? ""Resource restrictions include an illegal resource constraint combination - '{0-constraints}'","

Reply:
> line 32 : In " AuthenticatorNotFound", is this Conjur only? What about DAP? Also, I would say "... not configured/installed/enabled" but not "not implemented..."

this is for both conjur and dap but i think that it's ok to say `Conjur` as it is a component inside DAP. Also - it says "not implemented" because this is literally the case - the authenticator is not implemented in the code (e.g `authn-shuli`)

> Lines 58-59: "Resource type '{0}' is not a supported resource restriction. "
"The supported resources are '{1}'", "The supported resources are '{1}'",
Do you mean "The supported resource restrictions are....", or "The supported restrictions are..."

i mean "the supported resource". For example the resource "service-account".

> Line 47: "Status check not implemented for authenticator" What does "not implemented" mean? How about "Status of [authn-name] authenticator not verified."

It is literally not implemented in the code. not every authenticator has a status check.

> Line 53 - what does this mean, in plain english? ""Resource restrictions include an illegal resource constraint combination - '{0-constraints}'","

it means that, for example, a user set a host that has resource restrictions on both a system-assigned-identity and on a user-assigned-identity. this configuration doesn't make any sense and will never pass as no Azure AD token will have these characteristics. So we want to raise a configuration error.

Reply:
not implemnted (line 47: Then how about "There is not status check for ..." or Status check not supported for ...."

Reply:
i like it!

should it be `Status check not supported for authenticator` or `Status check is not supported for authenticator`?

Reply:
line 53 How about: Resource restrictions include an illegal combination of resource constraints - '{0-constraints}'"
It's easier to read (if it means the same thing...)

Reply:
good, thanks